tests/components/decora_wifi/test_config_flow.py

```diff
@@ -0,0 +1,144 @@
+"""Test the decora_wifi config flow."""
+from unittest.mock import patch
+
+from homeassistant import config_entries
+from homeassistant.components.decora_wifi.config_flow import CannotConnect
+from homeassistant.components.decora_wifi.const import DOMAIN
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+
+async def test_form(hass: HomeAssistant) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert not result["errors"]
+    with patch(
+        "homeassistant.components.decora_wifi.config_flow.DecoraWiFiSession.login",
+        return_value=True,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_USERNAME: "test-username",
+                CONF_PASSWORD: "test-password",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == FlowResultType.CREATE_ENTRY
+    assert result2["title"] == "test-username"
+    assert result2["data"] == {
+        CONF_USERNAME: "test-username",
+        CONF_PASSWORD: "test-password",
+    }
+
+
+async def test_form_invalid_auth(hass: HomeAssistant) -> None:
+    """Test we handle invalid auth."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.decora_wifi.config_flow.DecoraWiFiSession.login",
+        return_value=False,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_USERNAME: "test-username",
+                CONF_PASSWORD: "bad-password",
+            },
+        )
+
+    assert result2["type"] == FlowResultType.FORM
+    assert result2["errors"] == {"base": "invalid_auth"}
+
+
+async def test_form_cannot_connect(hass: HomeAssistant) -> None:
```

Comment:
Checkout `@pytest.mark.parametrize`. It's a great way to test a lot of these error -> error code tests. Checkout lastfm config flow tests for an example.

Reply:
fixed in https://github.com/home-assistant/core/pull/104573/commits/0a5c621cc1411d75b4ff6c16be609a0fbd8143f7